relay-server/src/actors/processor.rs

```diff
@@ -2888,6 +2895,81 @@ mod tests {
         (outcome_aggregator, test_store)
     }
 
+    #[tokio::test]
+    async fn test_dsc_respects_metrics_extracted() {
+        relay_test::setup();
+        let (outcome_aggregator, test_store) = services();
+
+        let config = Config::from_json_value(serde_json::json!({
+            "processing": {
+                "enabled": true,
+                "kafka_config": [],
+            }
+        }))
+        .unwrap();
+
+        let service: EnvelopeProcessorService = create_test_processor(config);
+
+        // Gets a ProcessEnvelopeState, either with or without the metrics_exracted flag toggled.
+        let get_state = |version: Option<u16>| {
+            let event = Event {
+                id: Annotated::new(EventId::new()),
+                ty: Annotated::new(EventType::Transaction),
+                transaction: Annotated::new("testing".to_owned()),
+                ..Event::default()
+            };
+
+            let mut project_state = state_with_rule_and_condition(
+                Some(0.0),
+                RuleType::Transaction,
+                SamplingMode::Received,
```

Comment:
nit: This `SamplingMode` parameter distracts from understanding the intention of the test, can we remove it from the helper function's signature?

Reply:
`state_with_rule_and_condition` is an old helper though, should i make a new one just to not include `SamplingMode`?

Reply:
I would just remove  that argument from the existing helper function. It is only called from two locations anyway.